src/evm/decoder.rs

```diff
@@ -90,11 +100,22 @@ impl TychoStreamDecoder {
             .insert(exchange.to_string(), decoder);
     }
 
+    /// Registers a filter function for a given exchange.
+    ///
+    /// The filter function is used to determine whether a given component should be included in
+    /// the state updates. This can be useful when only wanting to retrieve state updates for
+    /// components with a TVL higher than a given amount, for example.
     pub fn register_filter(&mut self, exchange: &str, predicate: FilterFn) {
         self.inclusion_filters
             .insert(exchange.to_string(), predicate);
     }
 
+    /// Decodes a `FeedMessage` into a `BlockUpdate` containing the updated states of protocol
+    /// components.
+    ///
+    /// The `BlockUpdate` will only contain updated protocol states for protocols components whose
+    /// protocols are registered, which pass the added inclusion filters, and whose token qualities
+    /// are above the minimum specified token quality.
```

Comment:
Imo should be mentioned first on struct level docs then mentioned briefly again here.